### ISSUE TITLE
Add Snyk scanner in the pipeline

### ISSUE THREAD

**Issue by jabrena** [jabrena/cursor-rules-java#840]
# User Story: Add Snyk Agent Scan to Skill Validation Pipeline

**As a** project maintainer
**I want to** run Snyk Agent Scan against the generated agent skills in CI
**So that** agent-skill security and supply-chain risks are detected automatically before changes are merged or released.

## Acceptance Criteria

See: Inline Gherkin scenarios below.

```gherkin
Feature: Snyk Agent Scan pipeline validation

Background:
Given the repository contains generated agent skills under .agents/skills
And the CI workflow generates the current skills from skills-generator before scanning

@acceptance-test
Scenario: Scan generated skills with Snyk Agent Scan in CI
Given the GitHub Actions workflow has access to the SNYK_TOKEN secret
When a push triggers the CI Builds workflow
Then the workflow runs Snyk Agent Scan against .agents/skills
And the scan runs in CI mode with explicit non-interactive trust flags
And the workflow fails when Snyk reports high-severity findings
And the workflow uploads a Snyk Agent Scan report artifact for inspection

@integration-test
Scenario: Fail clearly when SNYK_TOKEN is not configured
Given the GitHub Actions workflow does not have the SNYK_TOKEN secret
When the Snyk Agent Scan validation job runs
Then the job fails before invoking the scanner
And the log explains that SNYK_TOKEN is required

@integration-test
Scenario: Keep skill validation logs readable
Given the CI workflow needs to generate skills before scanner execution
When Maven runs in the skill validation jobs
Then Maven runs with --batch-mode and --no-transfer-progress
And the generated logs avoid unnecessary transfer progress noise
```

## Notes

- Scanner reference: https://github.com/snyk/agent-scan
- Implementation PR: #841
- The scanner requires a Snyk API token stored as the GitHub Actions secret `SNYK_TOKEN`.
- Current pipeline insights are tracked in the issue comments, including the first Snyk finding summary from the CI logs.

## INVEST Validation

- **Independent**: The Snyk scanner can be added as a separate CI validation job without depending on unrelated skill content changes.
- **Negotiable**: Scanner flags, reporting format, artifact naming, and failure policy can be refined while preserving the goal of automated Snyk validation.
- **Valuable**: Maintainers get automated visibility into prompt-injection, external-content, and skill supply-chain risks before release.
- **Estimable**: The scope is bounded to GitHub Actions workflow changes, README documentation updates, token configuration, and validation of scanner output.
- **Small**: The change fits in one iteration because it adds one scanner integration and supporting documentation.
- **Testable**: Completion is verified by CI running Snyk Agent Scan, failing on findings, and uploading a detailed scan report artifact.

**Comment by jabrena** [comment]
## Snyk Agent Scan Pipeline Insights

Source: [CI run 27403890358 / Snyk Agent Scan job](https://github.com/jabrena/cursor-rules-java/actions/runs/27403890358/job/80988121891)
Latest verification: [CI run 27407879122 / Snyk Agent Scan job](https://github.com/jabrena/cursor-rules-java/actions/runs/27407879122/job/81001453197)

The Snyk Agent Scan integration is running correctly. It scanned `99` generated skills and now passes with no remaining findings from the tracked table. The workflow also uploaded the `snyk-agent-scan-report` artifact for detailed inspection.

Pending issues to fix: `0`

| ID | Severity | Code | Skill | Risk | Insight from pipeline log | Suggested follow-up |
| ---: | --- | --- | --- | --- | --- | --- |
| ~~1~~ | ~~High~~ | ~~`W011`~~ | ~~`042-planning-openspec`~~ | ~~`0.75`~~ | ~~Reads supplied issues, plans, designs, ADRs, and OpenSpec artifacts; outsider-authored body text can enter the LLM context.~~ | ~~Fixed in `7895d8f8` by requiring sanitized summaries or explicit trust confirmation before reading raw outsider-authored body text.~~ |
| ~~2~~ | ~~High~~ | ~~`W012`~~ | ~~`161-java-profiling-detect`~~ | ~~`0.90`~~ | ~~Downloads async-profiler from GitHub at runtime and executes downloaded binaries/scripts.~~ | ~~Fixed in `7895d8f8` by removing the runtime download path and requiring a trusted preinstalled async-profiler distribution.~~ |
| ~~3~~ | ~~High~~ | ~~`W011`~~ | ~~`044-planning-jira`~~ | ~~`0.85`~~ | ~~Uses Jira issue descriptions and comments authored by other users as LLM input.~~ | ~~Fixed in `765a29c3` by keeping Jira analysis on list metadata plus user-provided sanitized summaries and removing raw description/comment retrieval paths.~~ |
| ~~4~~ | ~~High~~ | ~~`W011`~~ | ~~`523-frameworks-micronaut-testing-acceptance-tests`~~ | ~~`0.75`~~ | ~~Reads Gherkin `.feature` files and uses free-text `Feature`, `Scenario`, and step content to generate tests.~~ | ~~Fixed in `3551978d` by treating Gherkin scenario text as untrusted data, requiring trusted or sanitized scenario facts, and confirming selected scenarios before code generation.~~ |
| ~~5~~ | ~~High~~ | ~~`W011`~~ | ~~`041-planning-plan-mode`~~ | ~~`0.75`~~ | ~~Ingests supplied artifacts such as issues, PRs, wiki, or discussion threads.~~ | ~~Fixed in `224aa732` by requiring trusted planning inputs and sanitized summaries or explicit trust confirmation before reading raw external body text.~~ |
| ~~6~~ | ~~High~~ | ~~`W012`~~ | ~~`033-architecture-diagrams`~~ | ~~`0.70`~~ | ~~PlantUML/C4 examples use remote `!include` URLs fetched during rendering.~~ | ~~Fixed in `dd855b2a` by replacing remote C4/icon includes with trusted local PlantUML include paths and adding local-include constraints.~~ |
| ~~7~~ | ~~High~~ | ~~`W011`~~ | ~~`034-architecture-design-exploration`~~ | ~~`0.75`~~ | ~~Reads issue or requirement context that may be outsider-authored.~~ | ~~Fixed in `dd855b2a` by requiring sanitized design briefs or maintainer-provided requirement summaries before design exploration.~~ |
| ~~8~~ | ~~High~~ | ~~`W011`~~ | ~~`043-planning-github-issues`~~ | ~~`0.85`~~ | ~~Uses `gh issue view --json` / `--comments` to ingest GitHub issue bodies and comments.~~ | ~~Fixed in `bebf4810` by moving GitHub issue analysis to sanitized user-provided summaries and removing direct issue/milestone export ingestion.~~ |
| ~~9~~ | ~~High~~ | ~~`W012`~~ | ~~`703-technologies-fuzzing-testing`~~ | ~~`0.90`~~ | ~~Dockerfile downloads CATS from GitHub releases and executes it via `java -jar`.~~ | ~~Fixed in `fc0d0134` by requiring a trusted local `cats.jar` or approved prebuilt image instead of generated remote executable downloads.~~ |
| ~~10~~ | ~~High~~ | ~~`W011`~~ | ~~`014-agile-user-story`~~ | ~~`0.85`~~ | ~~Interactive questionnaire responses can contain outsider free text used to generate artifacts.~~ | ~~Fixed in `fc0d0134` by treating questionnaire answers and generated Gherkin as structured sanitized story data only.~~ |
| ~~11~~ | ~~Medium~~ | ~~`W021`~~ | ~~`200-agents-md`~~ | ~~N/A~~ | ~~Hidden Unicode characters detected.~~ | ~~Fixed in `f0de4617` by removing the hidden Unicode variation selector from the skill XML sources.~~ |

### Pattern Summary

| Pattern | Affected findings | Interpretation |
| --- | ---: | --- |
| ~~`W011` third-party content exposure~~ | ~~0 pending findings~~ | ~~Fixed by requiring sanitized summaries, trusted inputs, data-only treatment, and explicit authority boundaries.~~ |
| ~~`W012` external URL/runtime content~~ | ~~0 pending findings~~ | ~~Fixed for async-profiler, CATS, and C4 PlantUML include paths by using trusted local or preinstalled assets.~~ |
| ~~`W021` hidden Unicode~~ | ~~0 pending findings~~ | ~~Fixed in `f0de4617`; latest Snyk run no longer reports `200-agents-md` as a medium finding.~~ |
